engine_mgo.go

```diff
@@ -112,6 +112,9 @@ func (eng *mongoEngine) Load(collection Collection, id string) (bool, Document,
 		return false, Document{}, err
 	}
 	cleanup(content)
+	if eng.isBinaryId {
+		content["uuid"] = id
```

Comment:
this should not hard code "uuid" as the property name.  up-restorage can work with things other than uuid.